Dev Progress Bar

daman1321 opened the pull request:
Do not merge until reviewed

daman1321 committed:
Adds a progress bar to the player

Created a new progress bar to the player which will help user to drag the video to the different timestamp. This feature has been added in "media_panel.py" file.

Testing Steps
 1. Run then application main.py
 2. Click on the Load File from menu bar and run any video on the media player and try to drag th e video to different time frame.

Type: New Feature

Briggs-Richardson commented on View/media_panel.py:
```diff
@@ -19,11 +19,17 @@ def __init__(self):
         vertical_layout.addWidget(self.media_control_panel)
         self.setLayout(vertical_layout)
+
+        
```
Lines 34 and 35 should be removed.

There should only be one blank line at the end of a file.

Briggs-Richardson commented on View/media_panel.py:
```diff
@@ -19,11 +19,17 @@ def __init__(self):
-
+        
+        # create slider for the media player
+        self.slider = QSlider(PySide6.QtCore.Qt.Orientation.Horizontal)
```
Let's rename this data member to something more meaningful. Instead of slider, perhaps we should call it "progress_bar_slider"

Briggs-Richardson replied:
This change should be reflected in all usages of the data member as well.

Briggs-Richardson commented on Controllers/controller.py:
```diff
@@ -106,3 +111,17 @@ def open_file_dialog(self):
             self._media_player.play()
+
+    @Slot()
+    def position_changed(self):
```
Add a docstring for the position_changed function. Use the same style as the other docstrings (look at other function docstrings).

Briggs-Richardson commented on Controllers/controller.py:
```diff
@@ -106,3 +111,17 @@ def open_file_dialog(self):
+        self._window.media_panel.slider.setValue(position)
+
+    @Slot()
+    def duration_changed(self):
```
Add a docstring for the duration_changed function. Use the same style as the other docstrings (look at other function docstrings).

Briggs-Richardson replied:
This function should be placed above the open_file_dialog function. We organize the functions in alphabetical order.

Briggs-Richardson commented on Controllers/controller.py:
```diff
@@ -106,3 +111,17 @@ def open_file_dialog(self):
+        self._window.media_panel.slider.setRange(0, duration)
+
+    @Slot()
+    def set_position(self, position):
```
Add a docstring for the set_position function. Use the same style as the other docstrings (look at other function docstrings).

Briggs-Richardson commented on Controllers/controller.py:
```diff
@@ -106,3 +111,20 @@ def open_file_dialog(self):
+    @Slot()
+    def set_position(self, position):
+        """Command the video player to set the state of position while dragging"""
+        self._media_player.setPosition(position)
```
Add one new line at the end of the file.

Briggs-Richardson commented on Controllers/controller.py:
```diff
@@ -106,3 +111,20 @@ def open_file_dialog(self):
+        self._window.media_panel.progress_bar_slider.setValue(position)
+
+    @Slot()
+    def duration_changed(self):
```
The function duration_changed should be placed above the function open_file_dialog. Functions are defined in the file in alphabetical order.

Briggs-Richardson commented on Controllers/controller.py:
```diff
@@ -106,3 +111,20 @@ def open_file_dialog(self):
+
+    @Slot()
+    def set_position(self, position):
+        """Command the video player to set the state of position while dragging"""
```
Let's be more specific about what's dragging:

"""
Commands the video player to set the position state based on the value of the progress bar slider.
"""
"""

Briggs-Richardson commented on Controllers/controller.py:
```diff
@@ -106,3 +111,20 @@ def open_file_dialog(self):
+
+    @Slot()
+    def duration_changed(self):
+        """Command the video player to change the state when duration changed"""
```
This comment seems inaccurate. This function is changing the state of the progress bar, not the video player. Here is perhaps a better comment:
"""
Sets the range of the progress bar when the duration of the media player changes.
"""

Briggs-Richardson commented on Controllers/controller.py:
```diff
@@ -106,3 +111,20 @@ def open_file_dialog(self):
+
+    @Slot()
+    def position_changed(self):
+        """Command the video player to change state of the position"""
```
This comment seems inaccurate. The function is changing the value of the progress bar slider, not the video player. Here is perhaps a better comment:
"""
Sets the value of the progress bar slider based on the position of the media player.
"""

Briggs-Richardson commented on Controllers/controller.py:
```diff
@@ -106,3 +111,20 @@ def open_file_dialog(self):
             url = file_dialog.selectedUrls()[0]
             self._media_player.setSource(url)
             self._media_player.play()
```
Can we remove this line? Since we have a play/pause button, the video should no longer automatically play.

Briggs-Richardson committed:
Merge branch 'main' into dev_progress_bar

Briggs-Richardson committed:
Adds a progress bar to the player

Created a new progress bar to the player which will help user to drag the
video to the different timestamp. This feature has been added in "media_panel.py"
file.

Testing Steps
 1. Run then application main.py
 2. Click on the Load File from menu bar and run any video on the media player
    and try to drag the video to different time frame.

Type: New Feature
